### PR TITLE
Fix incorrect, but working test

### LINKED ISSUE #352: [news test] NanNew<RegExp>("foo", kNone) is not /foo/g ?
Hi,
the "news" test fails here (nodejs 0.10.33 with old v8 3.14) and just returns /foo/ not /foo/g.
https://github.com/nodejs/nan/blob/master/test/cpp/news.cpp#L87
https://github.com/nodejs/nan/blob/master/test/js/news-test.js#L54

is this failing for you too ?

### PR DESCRIPTION
**Pull request by kkoopa**
Fixes #352?

### CODE REVIEW

**Comment by bnoordhuis** [comment]
Not quite.  I think the issue is this:

```diff
diff --git a/test/js/news-test.js b/test/js/news-test.js
index 0de6e45..2a9c836 100644
--- a/test/js/news-test.js
+++ b/test/js/news-test.js
@@ -51,7 +51,7 @@ test('news', function (t) {
   t.equal(bindings.newLatin1String(), 'str<C3><AF>ng');
   t.equal(bindings.newUcs2String(), 'str<C3><AF>ng');
   t.equal(bindings.newStdString(), 'str<C3><AF>ng');
-  t.deepEquals(bindings.newRegExp(), /foo/g);
+  t.equals(bindings.newRegExp().toString(), '/foo/');
   t.deepEquals(bindings.newStringObject(), new String("foo"));
   t.deepEquals(bindings.newNumberObject(), new Number(0.5));
   t.deepEquals(bindings.newBooleanObject(), new Boolean(true));
```

**Comment by kkoopa** [comment]
Aren't they the same, sort of? Depending on which is seen as authoritative, the C++, part or the js part.

**Comment by kkoopa** [comment]
Ah, changing from deepEquals to equals too.... Why?

**Comment by bnoordhuis** [comment]
Sorry, I should have mentioned that: I think the bug is that t.deepEquals doesn't seem to value-compare regular expressions.  Changing it to t.equal doesn't really matter, that was just a consistency thing.

By the way, I updated the diff; apparently t.equal and t.equals are both legal.

**Comment by kkoopa** [comment]
How could the test ever fail then? Maybe with newer tap... Gonna check.

**Comment by kkoopa** [comment]
Nope, but updating tap caught something else. I have made the changes.

File: test/js/news-test.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ test('news', function (t) {
   t.equal(bindings.newLatin1String(), 'strïng');
   t.equal(bindings.newUcs2String(), 'strïng');
   t.equal(bindings.newStdString(), 'strïng');
-  t.deepEquals(bindings.newRegExp(), /foo/g);
+  t.equal(bindings.newRegExp(), /foo/);
```
Comment (bnoordhuis) [inline test/js/news-test.js]: Does that actually work?  t.equal is identity equality, isn't it?

Comment (kkoopa) [inline test/js/news-test.js]: https://github.com/isaacs/node-tap#equalfound-wanted-message-extra

**Comment by kkoopa** [comment]
There, Travis seems to like this.

**Comment by bnoordhuis** [comment]
LGTM

**Comment by kkoopa** [comment]
Thanks.

**Pull request merged**